training/run_pseudo_labelling.py

```diff
@@ -796,17 +796,30 @@ def compute_metrics(preds, labels, file_ids):
         # we do not want to group tokens when computing the metrics
         label_str = tokenizer.batch_decode(labels, skip_special_tokens=True)
 
-        # normalize everything and re-compute the WER
-        norm_pred_str = [normalizer(pred) for pred in pred_str]
-        norm_label_str = [normalizer(label) for label in label_str]
-        # for logging, we need the pred/labels to match the norm_pred/norm_labels, so discard any filtered samples here
-        pred_str = [pred_str[i] for i in range(len(norm_pred_str)) if len(norm_label_str[i]) > 0]
-        label_str = [label_str[i] for i in range(len(norm_label_str)) if len(norm_label_str[i]) > 0]
-        file_ids = [file_ids[i] for i in range(len(file_ids)) if len(norm_label_str[i]) > 0]
-        # filtering step to only evaluate the samples that correspond to non-zero normalized references:
-        norm_pred_str = [norm_pred_str[i] for i in range(len(norm_pred_str)) if len(norm_label_str[i]) > 0]
-        norm_label_str = [norm_label_str[i] for i in range(len(norm_label_str)) if len(norm_label_str[i]) > 0]
```

Comment:
These lines only keep the `norm_pred_str` (reference) and `norm_label_str` (target) where the `norm_label_str` is not empty. Having an empty reference is valid. In this case, for a reference set of N words we have:
*  N deletions (as many deletions as we do number of words in our reference set)
* 0 substitutions
* 0 insertions

So the WER is: (N + 0 + 0) / N = 1

You can see this with a toy example:
```py
from jiwer import wer

reference = "hello world"
hypothesis = ""

error = wer(reference, hypothesis)
print(error)
```
**Print Output:**
```
1.0
```

=> so there shouldn't be a need to have an additional check for empty normalised predictions! These should be valid in the WER calculation. Let me know if you have a minimal repro to rebuttal this!